rust/main/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -267,28 +276,40 @@ impl IsmCachePolicyClassifier {
         root_ism: H256,
         domain: &HyperlaneDomain,
         ism_module_type: ModuleType,
+        app_context: Option<&String>,
     ) -> IsmCachePolicy {
-        let default_ism = match self.default_ism_getter.get().await {
-            Ok(default_ism) => default_ism,
-            Err(err) => {
-                tracing::warn!(?err, "Error fetching default ISM for ISM cache policy, falling back to default cache policy");
-                return IsmCachePolicy::default();
+        for config in &self.ism_cache_configs {
+            let matches_module = match &config.selector {
+                IsmCacheSelector::DefaultIsm => {
+                    let default_ism = match self.default_ism_getter.get().await {
+                        Ok(default_ism) => default_ism,
+                        Err(err) => {
+                            tracing::warn!(?err, "Error fetching default ISM for ISM cache policy, falling back to default cache policy");
+                            return IsmCachePolicy::default();
```

Comment:
what if fetching the default ISM fails here, but another cache config selector would've matched? Does this stop searching for a match too soon? Since we return `IsmCachePolicy::default()` at the end of this function, maybe we could `continue` on error